reference/FungibleToken.scilla

```diff
@@ -0,0 +1,429 @@
+  is_owner = builtin eq _sender contract_owner;
+    match is_owner with
+    | False =>
+      err = CodeNotAuthorized;
+      MakeError err
+    | True =>
+      ProcedureBurn burn_account amount
+    end
+end
+
+(* @dev: Make an address an operator of the caller.                           *)
+(* @param operator: Address to be set as operator. Cannot be calling address. *)
+transition AuthorizeOperator(operator: ByStr20)
+  is_sender = builtin eq operator _sender;
+  match is_sender with
+  | True =>
+    (* Operator is authorizing self, return error code *)
+    err = CodeNotAuthorized;
+    MakeError err
+  | False =>
+    authorize = True;
+    operators_map[_sender][operator] := authorize;
+    e = {_eventname : "AuthorizeOperatorSuccess"; operator : operator};
+    event e
+  end
+end
+
+(* @dev: Revoke an address from being an operator of the caller. *)
+(* @param operator: Address to be removed as operator.           *)
+transition RevokeOperator(operator: ByStr20)
+  get_operator <- operators_map[_sender][operator];
+  match get_operator with
+  | None =>
+    (* Operator to be removed not found *)
+    err = CodeNotFound;
+    MakeError err
+  | Some status =>
+    delete operators_map[_sender][operator];
+    e = {_eventname : "RevokeOperatorSuccess"; operator : operator};
+    event e
+  end
+end
+
+(* @dev: Revoke a default operator from being an operator of the caller *)
+(* @param operator:  Address of the default operator to be revoked.     *)
+transition RevokeDefaultOperator(operator : ByStr20)
+  is_default_operator =  is_default_operator f_eq operator default_operators;
+  match is_default_operator with
+  | False =>
+    err = CodeNotFound;
+    MakeError err
+  | True  => 
+    verdad = True;
+    revoked_default_operators[_sender][operator] := verdad;
+    e = {_eventname : "RevokedDefaultOperatorSuccess"; revoked_operator : operator; revoker : _sender};
+    event e
+    end
+end
+
+(* @dev: Re-authorize a default operator as an operator of the caller    *)
+(* @param operator: Address of the default operator to be re-authorized. *)
+transition ReAuthorizeDefaultOperator(operator : ByStr20)
+is_default_operator = is_default_operator f_eq operator default_operators;
+  match is_default_operator with
+  | False =>
+    err = CodeNotFound;
+    MakeError err
+  | True  => 
+    delete revoked_default_operators[_sender][operator];
+    e = { _eventname : "ReAuthorizedDefaultOperatorSuccess"; revoked_operator : operator; revoker : _sender};
+    event e
+  end
+end
+
+(* @dev: Returns true if an address is an operator or default operator of a token_owner. *)
+(* @param operator:    Address of a potential operator.                                  *)
+(* @param token_owner: Address of a token_owner.                                         *)
+transition IsOperatorFor(token_owner: ByStr20, operator: ByStr20)
+  get_operator <- operators_map[token_owner][operator];
+  is_operator_approved = 
```

Comment:
I saw the same logic in another transition before, maybe we can use a procedure for this?

Reply:
As procedure does not return a value, I cannot use a procedure here as the value that is needed is a `Bool` that will be needed for the next computation.